Source/Dafny/JsonDiagnostics.cs

```diff
@@ -0,0 +1,114 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using Microsoft.Boogie;
+using VCGeneration;
+
+namespace Microsoft.Dafny;
+
+record DiagnosticMessageData(MessageSource source, ErrorLevel level, IToken tok, string? category, string message, List<ErrorInformation.AuxErrorInfo>? related) {
+  private static JsonObject SerializePosition(IToken tok) {
+    return new JsonObject {
+      ["line"] = tok.line,
+      ["character"] = tok.col - 1
```

Comment:
Why put a character indexed by zero, but a line indexed by 1?
Let's be consistent
```suggestion
      ["character"] = tok.col
```

Reply:
That's what the CLI does, and we want the same output in JSON and non-json mode.